From fa4f89078f3cc3ec6eb41cff2580a63f3ba28a45 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 25 Feb 2016 03:35:52 -0500
Subject: [PATCH] wr-policy: allow daemon write to socket file

Reference sources: selinux-policy-3.13.1-128.21.fc22.src.rpm
Changes come from: policy-f22-base.patch

* allow daemon write to init_var_run_t socket file
* allow daemon send message to unix_dgram_socket

Fix avc denials like:

  avc: denied { write } for pid=363 comm="avahi-daemon" \
  name="notify" dev="tmpfs" ino=8182 \
  scontext=system_u:system_r:avahi_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:init_var_run_t:s0 \
  tclass=sock_file permissive=1

  avc: denied { sendto } for pid=363 comm="avahi-daemon" \
  path="/run/systemd/notify" \
  scontext=system_u:system_r:avahi_t:s0-s15:c0.c1023 \
  tcontext=system_u:system_r:init_t:s0-s15:c0.c1023 \
  tclass=unix_dgram_socket permissive=1

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/init.te | 4 ++++
 1 file changed, 4 insertions(+)

--- refpolicy.orig/policy/modules/system/init.te
+++ refpolicy/policy/modules/system/init.te
@@ -195,6 +195,10 @@ seutil_read_config(init_t)
 
 miscfiles_read_localization(init_t)
 
+allow daemon init_t:unix_dgram_socket sendto;
+# need write to /var/run/systemd/notify
+init_write_pid_socket(daemon)
+
 ifdef(`init_systemd',`
 	# handle instances where an old labeled init script is encountered.
 	typeattribute init_t init_run_all_scripts_domain;
